plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/parameterInfo/KotlinHighLevelFunctionParameterInfoHandler.kt

```diff
@@ -115,7 +115,7 @@ abstract class KotlinHighLevelParameterInfoWithCallHandlerBase<TArgumentList : K
             }
 
             val candidates = resolvedCall?.targetFunction?.candidates ?: return null
-            context.itemsToShow = candidates.map { CandidateInfo(it.createPointer()) }.toTypedArray()
+            context.itemsToShow = candidates.map { CandidateInfo(it.createPointer(), it.deprecationStatus != null) }.toTypedArray()
```

Comment:
Please consider hiding the entry if the deprecation level is `HIDDEN`. Looks like FE1.0 doesn't show any deprecated member of this level.

Reply:
If deprecation level is `HIDDEN`, in FE 1.0, it comes out as `UNRESOLVED_REFERENCE` (the candidate is hidden for all practical purposes) but in FIR it's `INVISIBLE_REFERENCE` with a candidate and hence we show it in the popup. There are a few ways to fix this:

1. Filter out the candidate here in the handler.
2. Filter out candidates from `ConeHiddenCandidateError` in HL API.
3. Change the diagnostic to `UNRESOLVED_REFERENCE` for `HIDDEN` deprecation level.

My first inclination is to go with the last option since that matches FE 1.0 behavior. Will ask offline on whether we should do that or not.

Reply:
Updated KOTLIN-MR-188 so that candidates from `ConeHiddenCandidateError` are not returned. Also added more tests here.

Reply:
Updated this PR so we're filtering candidates hidden due to visibility here on the client. Left handling of `@Deprecated(HIDDEN)` to a future change.